brave/src/main/java/com/linecorp/decaton/processor/runtime/BraveTracingProvider.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright 2020 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.decaton.processor.runtime;
+
+
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+
+import brave.Span;
+import brave.kafka.clients.KafkaTracing;
+import brave.propagation.CurrentTraceContext.Scope;
+
+public class BraveTracingProvider implements TracingProvider<Span> {
+    private static final ThreadLocal<Scope> CURRENT_SCOPE = new ThreadLocal<>();
+    private final KafkaTracing kafkaTracing;
+
+    public BraveTracingProvider(KafkaTracing kafkaTracing) {
+        this.kafkaTracing = kafkaTracing;
+    }
+
+    @Override
+    public Span traceFor(ConsumerRecord<?, ?> record) {
+        return kafkaTracing.nextSpan(record).name("decaton.process").start();
```

Comment:
"decaton.process" sounds like to correspond for `DecatonProcessor#process` method call rather than the entire flow in decaton consumer doesn't it?
To me below naming are the most natural:
* traceFor => decaton.consume (or just "decaton"?)
* processingStart => decaton.process(or).start
* processingReturn => decaton.process(or).return

WDYT?

Reply:
Will rename the span to `decaton`.
I didn't want to say `decaton.process.start` because I realised this will actually only call `start`/`return` once for the whole `ProcessPipeline`, which might actually include multiple processors.